msal/src/main/java/com/microsoft/identity/client/PublicClientApplication.java

```diff
@@ -712,6 +712,40 @@ public void acquireToken(@NonNull final Activity activity,
         );
     }
 
+    /**
+     * Acquire token interactively, will pop-up webUI. Interactive flow will skip the cache lookup.
+     * Default value for {@link UiBehavior} is {@link UiBehavior#SELECT_ACCOUNT}.
+     *
+     * @param activity    Non-null {@link Activity} that is used as the parent activity for launching the {@link AuthenticationActivity}.
+     *      *                 All apps doing an interactive request are required to call the
+     *      *                 {@link PublicClientApplication#handleInteractiveRequestRedirect(int, int, Intent)} within the calling
+     *      *                 activity {@link Activity#onActivityResult(int, int, Intent)}.
+     * @param resource    The non-null string of resource identifier.
+     * @param callback    The {@link AuthenticationCallback} to receive the result back.
+     *      *                 1) If user cancels the flow by pressing the device back button, the result will be sent
+     *      *                 back via {@link AuthenticationCallback#onCancel()}.
+     *      *                 2) If the sdk successfully receives the token back, result will be sent back via
+     *      *                 {@link AuthenticationCallback#onSuccess(IAuthenticationResult)}
+     *      *                 3) All the other errors will be sent back via
+     *      *                 {@link AuthenticationCallback#onError(MsalException)}.
+     */
+    public void acquireToken(@NonNull final Activity activity,
+                             @NonNull final String resource,
+                             @NonNull final AuthenticationCallback callback) {
+        acquireToken(
+                activity,
+                mapResourceToDefaultScope(resource),
```

Comment:
Although we've tagged the `resource` parameter here as `@NonNull` I think we shouldn't trust the end-developer to adhere... can we null-check this param before we call `acquireToken()`?